### PR TITLE
Remove python-keycloak

### PR DESCRIPTION
**Pull request by FedericoNegri**
Background:
- `python-keycloak` versions < 3.9.1 depend on `python-jose` which in turn depends on `ecdsa`, which has safety vulnerabilities
- because of the above, `python-keycloak >= 3.9.1` [replaced](https://github.com/marcospereirampj/python-keycloak/pull/531) `python-jose` with `jwcrypto` which is LGPL. 

I propose to get rid of `python-keycloak` as a PyHPS dependency (we'd only keep using it for tests). The package was only used in the `AuthApi` as a convenience to query/create/modify/delete users as well as query their groups and roles.

Since by default HPS users do not have the realm management `manage-users` role, it seems just misleading to expose the create/modify/delete users endpoints. I therefore removed those and rather added an example in the doc showing how to use the `KeycloakAdmin` client from `python-keycloak` to do so.

Nightly build run with Python 3.9 - 3.12 tests: https://github.com/ansys/pyhps/actions/runs/8158129451

### Breaking change

`ansys.hps.client.auth.authenticate` moved to `ansys.hps.client.authenticate` to avoid circular dependencies

### CODE REVIEW

**Review by nezgrath — APPROVED** [review]

File: src/ansys/hps/client/auth/api/auth_api.py [hunk 1]
```diff
@@ -21,20 +21,16 @@
...
 from ..schema.user import UserSchema
 
 
 class AuthApi:
-    """Provides the Python interface to the Authorization Service API.
-
-    Admin users with the Keycloak "manage-users" role can create
-    users as well as modify or delete existing users. Non-admin users are only allowed
-    to query the list of existing users.
+    """Minimal wrapper around the Keycloak API to query users info.
```
Comment (PipKat) [inline src/ansys/hps/client/auth/api/auth_api.py:33]:
```suggestion
    """Provides a minimal wrapper around the Keycloak API to query user information.
```
All class descriptions should start with a verb ending in "s" or "es" for consistency (and per our standard).

File: src/ansys/hps/client/auth/api/auth_api.py [hunk 2]
```diff
@@ -93,19 +78,48 @@ def get_users(self, as_objects=True, **query_params) -> List[User]:
 
         For a list of supported query parameters, see the Keycloak API documentation.
```
Comment (PipKat) [inline src/ansys/hps/client/auth/api/auth_api.py:79]: Perhaps make "Keycloak API documentation" a link?

**Review by PipKat — APPROVED** [review]
@FedericoNegri Only two comments, with only one of those being a correction! This approval is only on doc content--but what you did was excellent. I've never only had one suggested correction, even on PRs much smaller than this one!

**Comment by FedericoNegri** [comment]
> @FedericoNegri Only two comments, with only one of those being a correction! This approval is only on doc content--but what you did was excellent. I've never only had one suggested correction, even on PRs much smaller than this one!

@PipKat your comment made we wonder and I realized I forgot to include the new "managing users" doc guide! please review this one https://github.com/ansys/pyhps/pull/352/files#diff-f835861f0b7fc9c9c1098462eb63089cd78a308d47e973efa1fb1037eaaf98a9, I'm sure you'll have more corrections :-)

File: doc/source/getting_started/index.rst [hunk 3]
```diff
@@ -46,19 +46,16 @@ To install PyHPS in user mode, run this command:
...
-``pip`` automatically installs these package dependencies in case they are not already installed:
+Dependencies
+~~~~~~~~~~~~
 
-- requests_
-- marshmallow_
-- marshmallow_oneofschema_
-- python-keycloak_
-- pydantic_
+The following packages are required dependencies of PyHPS:
```
Comment (PipKat) [inline doc/source/getting_started/index.rst:52]:
```suggestion
PyHPS requires these packages as dependencies:
```
Rewriting to active voice.

File: doc/source/user_guide/managing_users.rst [hunk 4]
```diff
@@ -0,0 +1,130 @@
+Managing Users
```
Comment (PipKat) [inline doc/source/user_guide/managing_users.rst:1]:
```suggestion
Managing users
```

File: doc/source/user_guide/managing_users.rst [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    from ansys.hps.client.auth import AuthApi, User
+
+    cl = Client(
+        url="https://127.0.0.1:8443/hps", username="repuser", password="repuser"
+    )
+
+    auth_api = AuthApi(cl)
+    users = auth_api.get_users(firstName="john", exact=False)
+
+You can also use it to find the email of the user that created and last modified a resource in JMS, for instance a job:
```
Comment (PipKat) [inline doc/source/user_guide/managing_users.rst:25]:
```suggestion
You can also find the email of the user that created and last modified a resource in JMS, for instance a job:
```

File: doc/source/user_guide/managing_users.rst [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    )
+
+
+Using ``python-keycloak`` to modify and create users
+----------------------------------------------------
+
+Administrative users with the Keycloak realm management ``manage-users`` role
+can create, modify, and delete users.
+This functionality is not exposed in the :exc:`AuthApi <ansys.hps.client.auth.api.AuthApi>`.
+However, third-party Python packages can be used.
```
Comment (PipKat) [inline doc/source/user_guide/managing_users.rst:55]:
```suggestion
This functionality is not exposed in :exc:`AuthApi <ansys.hps.client.auth.api.AuthApi>`.
However, you can use third-party Python packages. 
```

File: doc/source/user_guide/managing_users.rst [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+This functionality is not exposed in the :exc:`AuthApi <ansys.hps.client.auth.api.AuthApi>`.
+However, third-party Python packages can be used. 
+
+The following examples show how to modify and create users using the
+`python-keycloak <https://pypi.org/project/python-keycloak/>`_ package.
+
+Connect to the Keycloak Admin API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Connecting as Keycloak administrator (using default credentials) gives you full control:
```
Comment (PipKat) [inline doc/source/user_guide/managing_users.rst:63]:
```suggestion
Connecting as a Keycloak administrator (using default credentials) gives you full control:
```

File: doc/source/user_guide/managing_users.rst [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    keycloak_client = KeycloakAdmin(
+        server_url="https://localhost:8443/hps/auth/",
+        username="keycloak",
+        password="keycloak123",
+        realm_name="rep",
+        user_realm_name="master",
+        verify=False,
+    )
+
+Alternatively, you can also connect as a regular user:
```
Comment (PipKat) [inline doc/source/user_guide/managing_users.rst:78]:
```suggestion
Alternatively, you can connect as a regular user:
```

File: doc/source/user_guide/managing_users.rst [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            "type": "password",
+            "value": "my_new_password",
+        }
+    ]
+    keycloak_client.update_user(user["id"], user)
+
+
+Create a new user with a temporary password
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (PipKat) [inline doc/source/user_guide/managing_users.rst:110]:
```suggestion
Create a user with a temporary password
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by PipKat — APPROVED** [review]
Minor edits suggested.

**Pull request merged**